allennlp/data/semparse/worlds/nlvr_world.py

```diff
@@ -0,0 +1,436 @@
+                            target_attribute: AttributeType) -> Set[Box]:
+        return cls._filter(set_to_filter, attribute_function, target_attribute, operator.le)
+
+    ## Object filtering functions
+    @classmethod
+    def black(cls, objects_set: Set[Object]) -> Set[Object]:
+        return cls._filter(objects_set, cls._get_single_color, 'black', operator.eq)
+
+    @classmethod
+    def blue(cls, objects_set: Set[Object]) -> Set[Object]:
+        return cls._filter(objects_set, cls._get_single_color, 'blue', operator.eq)
+
+    @classmethod
+    def yellow(cls, objects_set: Set[Object]) -> Set[Object]:
+        return cls._filter(objects_set, cls._get_single_color, 'yellow', operator.eq)
+
+    @classmethod
+    def circle(cls, objects_set: Set[Object]) -> Set[Object]:
+        return cls._filter(objects_set, cls._get_single_shape, 'circle', operator.eq)
+
+    @classmethod
+    def square(cls, objects_set: Set[Object]) -> Set[Object]:
+        return cls._filter(objects_set, cls._get_single_shape, 'square', operator.eq)
+
+    @classmethod
+    def triangle(cls, objects_set: Set[Object]) -> Set[Object]:
+        return cls._filter(objects_set, cls._get_single_shape, 'triangle', operator.eq)
+
+    @classmethod
+    def same_color(cls, objects_set: Set[Object]) -> Set[Object]:
+        """
+        Returns the set of objects whose color is the most frequent color in the initial set, if the highest
+        frequency is greater than 1. If not, all objects are of different colors, and this method returns a
+        null set.
+        """
+        all_colors = [cls._get_single_color(obj) for obj in objects_set]
+        color_counts: Dict[str, int] = defaultdict(int)
+        for _color in all_colors:
+            color_counts[_color] += 1
+        if max(color_counts.values()) <= 1:
+            return set()
+        most_frequent_color = max(all_colors, key=color_counts.get)
+        return cls._filter(objects_set, cls._get_single_color, most_frequent_color, operator.eq)
+
+    @classmethod
+    def same_shape(cls, objects_set: Set[Object]) -> Set[Object]:
+        """
+        Returns the set of objects whose shape is the most frequent shape in the initial set, if the highest
+        frequency is greater than 1. If not, all objects are of different shapes, and this method returns a
+        null set.
+        """
+        all_shapes = [cls._get_single_shape(obj) for obj in objects_set]
+        shape_counts: Dict[str, int] = defaultdict(int)
+        for _shape in all_shapes:
+            shape_counts[_shape] += 1
+        if max(shape_counts.values()) <= 1:
+            return set()
+        most_frequent_shape = max(all_shapes, key=shape_counts.get)
+        return cls._filter(objects_set, cls._get_single_shape, most_frequent_shape, operator.eq)
+
+    @classmethod
+    def touch_bottom(cls, objects_set: Set[Object]) -> Set[Object]:
+        return cls._filter(objects_set, lambda x: x.get_attribute("y_loc"), 0, operator.eq)
```

Comment:
No need for this fancy filter.  Just as in `black`, `triangle`, etc.: `return [object for object in objects if object.y_loc == 0]`.

Reply:
Done.